sklearn/model_selection/_search.py

```diff
@@ -653,6 +653,19 @@ def fit(self, X, y=None, groups=None, **fit_params):
         if self.return_train_score:
             train_scores = _aggregate_score_dicts(train_score_dicts)
 
+        # check for inf or -inf values in the train and test scores.
+        for test_score in test_scores['score']:
```

Comment:
There won't be such a key as 'score' if multiple metrics are used.

Also, you could use np.any and no.isfinite on the array